daemon/api_notices.go

```diff
@@ -197,6 +197,9 @@ func getNotice(c *Command, r *http.Request, user *auth.UserState) Response {
 	if notice == nil {
 		return NotFound("cannot find notice with id %q", noticeID)
 	}
+	if err := checkCanAccessNotice(notice, c.d, r, user); err != nil {
```

Comment:
this is too late to change our mind if the request came from snapd-snap.socket, no?  we cannot avoid to do something about the main access checker for this

Reply:
Moved the logic into the main read access checker, Thanks!